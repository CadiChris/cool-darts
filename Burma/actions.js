import Contrats from "./Contrats";

export const INSCRIRE_JOUEUR = "INSCRIRE_JOUEUR";
export const inscrireJoueur = nomDuJoueur => ({
  type: INSCRIRE_JOUEUR,
  payload: {
    nomDuJoueur
  }
});

export const DEMARRER_PARTIE = "DEMARRER_PARTIE";
export const demarrerPartie = () => ({
  type: DEMARRER_PARTIE
});

export const VOLEE = "VOLEE";
export const voleeSurChiffre = (lanceur, chiffre, nombreDeTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.chiffre(chiffre, nombreDeTouches)
  }
});

export const voleeSurDouble = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.double(chiffresTouches)
  }
});

export const voleeSurTriple = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.triple(chiffresTouches)
  }
});

export const voleeSurBull = (
  lanceur,
  nombreDeSimplesBull,
  nombreDeDoublesBull
) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.bull(nombreDeSimplesBull, nombreDeDoublesBull)
  }
});
