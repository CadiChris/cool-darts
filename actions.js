const INSCRIRE_JOUEUR = "INSCRIRE_JOUEUR";
const inscrireJoueur = nomDuJoueur => ({
  type: INSCRIRE_JOUEUR,
  payload: {
    nomDuJoueur
  }
});

const DEMARRER_PARTIE = "DEMARRER_PARTIE";
const demarrerPartie = () => ({
  type: DEMARRER_PARTIE
});

const VOLEE_SUR_CHIFFRE = "VOLEE_SUR_CHIFFRE";
const voleeSurChiffre = (lanceur, chiffre, nombreDeTouches) => ({
  type: VOLEE_SUR_CHIFFRE,
  payload: {
    lanceur,
    chiffre,
    nombreDeTouches
  }
});

const VOLEE_SUR_DOUBLE = "VOLEE_SUR_DOUBLE";
const voleeSurDouble = (lanceur, chiffresTouches) => ({
  type: VOLEE_SUR_DOUBLE,
  payload: {
    lanceur,
    chiffresTouches
  }
});

const VOLEE_SUR_TRIPLE = "VOLEE_SUR_TRIPLE";
const voleeSurTriple = (lanceur, chiffresTouches) => ({
  type: VOLEE_SUR_TRIPLE,
  payload: {
    lanceur,
    chiffresTouches
  }
});

const VOLEE_SUR_BULL = "VOLEE_SUR_BULL";
const voleeSurBull = (lanceur, nombreDeSimplesBull, nombreDeDoublesBull) => ({
  type: VOLEE_SUR_BULL,
  payload: {
    lanceur,
    nombreDeSimplesBull,
    nombreDeDoublesBull
  }
});

export {
  INSCRIRE_JOUEUR,
  inscrireJoueur,
  DEMARRER_PARTIE,
  demarrerPartie,
  VOLEE_SUR_CHIFFRE,
  VOLEE_SUR_DOUBLE,
  VOLEE_SUR_TRIPLE,
  VOLEE_SUR_BULL,
  voleeSurChiffre,
  voleeSurDouble,
  voleeSurTriple,
  voleeSurBull
};
