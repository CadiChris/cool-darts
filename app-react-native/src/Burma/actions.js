import Contrats from "./arbitrage/Contrats";

export const DEMARRER_PARTIE = "DEMARRER_PARTIE";
export const demarrerPartie = joueurs => ({
  type: DEMARRER_PARTIE,
  joueurs
});

export const VOLEE = "VOLEE";
export const voleeSurChiffre = (lanceur, chiffre, nombreDeTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.chiffre(chiffre, nombreDeTouches)
  }
});

export const voleeSurDouble = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.double(chiffresTouches)
  }
});

export const voleeSurTriple = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.triple(chiffresTouches)
  }
});

export const voleeSurBull = (lanceur, nombreDeBulls) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.bull(nombreDeBulls)
  }
});
