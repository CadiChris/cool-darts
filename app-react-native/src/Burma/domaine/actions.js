import Contrats from './arbitrage/Contrats';

export const DEMARRER_BURMA = 'BURMA/DEMARRER';
export const demarrerBurma = joueurs => ({
  type: DEMARRER_BURMA,
  joueurs,
});

export const VOLEE = 'BURMA/VOLEE';
export const voleeSurChiffre = (lanceur, chiffre, nombreDeTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.chiffre(chiffre, nombreDeTouches),
  },
});

export const voleeSurDouble = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.double(chiffresTouches),
  },
});

export const voleeSurTriple = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.triple(chiffresTouches),
  },
});
