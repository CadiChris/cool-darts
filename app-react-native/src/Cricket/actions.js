export const demarrerPartie = joueurs => ({
  type: "DEMARRER_PARTIE",
  joueurs
});

export const lancerFlechette = (joueur, chiffre, touches) => ({
  type: "LANCER_FLECHETTE",
  joueur,
  chiffre,
  touches
});

export const nouvellePartie = () => ({
  type: "NOUVELLE_PARTIE"
});
