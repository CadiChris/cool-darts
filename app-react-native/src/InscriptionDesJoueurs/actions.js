export const INSCRIRE_JOUEUR = "INSCRIRE_JOUEUR";
export const inscrireJoueur = nomDuJoueur => ({
  type: INSCRIRE_JOUEUR,
  nomDuJoueur
});

export const DESINSCRIRE_JOUEUR = "DESINSCRIRE_JOUEUR";
export const desinscrireJoueur = nomDuJouer => ({
  type: DESINSCRIRE_JOUEUR,
  nomDuJouer
});

export const CHOISIR_JEU = "CHOISIR_JEU";
export const choisirJeu = jeuChoisi => ({
  type: CHOISIR_JEU,
  jeuChoisi
});
