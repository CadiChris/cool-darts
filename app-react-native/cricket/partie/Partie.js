import React from 'react'
import { connect } from 'react-redux'
import { View, Text } from 'react-native'
import Lanceur from './Lanceur'
import InscriptionDesJoueurs from './InscriptionDesJoueurs'

class Partie extends React.Component {
  constructor(props) {
    super(props)
  }

  render() {
    const {peutDemarrer, phase} = this.props
    console.log(phase)
    return (
      <View>
        {
          phase === 'INSCRIPTION'
            ? <InscriptionDesJoueurs />
            : <Lanceur lanceur={{nom: 'Christophe'}}/>

        }
        <Text>{peutDemarrer ? 'READY' : 'Pas encore'}</Text>
      </View>
    )
  }
}

function mapStateToProps(state) {
  console.log(state)
  return {
    peutDemarrer: state.partie.peutDemarrer,
    phase: state.partie.phase
  }
}

export default connect(mapStateToProps)(Partie)