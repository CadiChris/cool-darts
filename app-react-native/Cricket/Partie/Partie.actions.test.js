import { inscrireJoueur } from './Partie.actions'

it('inscrit un joueur', () => {
  expect(inscrireJoueur('J1')).toMatchSnapshot()
})