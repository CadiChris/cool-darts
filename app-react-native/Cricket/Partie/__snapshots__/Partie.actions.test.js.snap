// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inscrit un joueur 1`] = `
Object {
  "joueur": "J1",
  "type": "INSCRIRE_JOUEUR",
}
`;
